src/Knp/Menu/Matcher/Voter/RouteVoter.php

```diff
@@ -91,4 +99,25 @@ private function isMatchingRoute(Request $request, array $testedRoute): bool
 
         return true;
     }
+
+    /**
+     * @phpstan-param array{route?: string|null, pattern?: string|null, parameters?: array<string, mixed>, query_parameters?: array<string, mixed>} $testedRoute
```

Comment:
for query parameters, it should be `array<string, string>`, as the query string can only contain strings.